agent-inject/agent/agent_test.go

```diff
@@ -66,19 +66,25 @@ func TestValidate(t *testing.T) {
 	}{
 		{
 			Agent{
-				Namespace:          "test",
-				ServiceAccountPath: "foobar",
-				ServiceAccountName: "foobar",
-				ImageName:          "test",
-				ConfigMapName:      "test",
+				Namespace: "test",
+				ServiceAccountTokenVolume: ServiceAccountTokenVolume{
```

Comment:
It looks like these all need to be type `*ServiceAccountTokenVolume`.

```suggestion
				ServiceAccountTokenVolume: &ServiceAccountTokenVolume{
```